File: Algorithm - Measurement and Calculation for Residential Roof.jl

# Author: PHNO - Tecnólogo | Pós-Graduado
# Release Date: 18/11/2024
# Version: 1.0.1v
# Replit: @PHNO, @PHREPLIT
# E-mail: phreplit@gmail.com

# Algorithm: Measurement and Calculation for Residential Roof, with menu and console compilation.

println(" ") 
println(" Measurement and Calculation for Residential Roof ")
println("")
println("")
println("1 - Calculate Square Meter of Roof [4 equal sides]")
println("2 - Calculate the quantity of American tiles per Square Meter.")
println("3 - Calculate the quantity of colonial tiles per Square Meter.")
println("4 - Calculate the quantity of Italian tiles per Square Meter.")
println("5 - Calculate the quantity of Portuguese tiles per Square Meter.")
println("6 - Calculate the quantity of Roman tiles per Square Meter.")
println("7 - Exit")
println("[8] Info")
println("[9] About")
println(" ") 

println("Enter with your choice: ") # choice - option
   op = readline() 
   op = parse(Int64, op) 

if op == 1
   println("")
   println("Enter the roof length: ") 
   v1 = readline() 
   println("")
   println("Enter the roof width: ") 
   v2 = readline() 
   result = v1 * v2
   println("")
   println("Square meters are equivalent to: ", result)
   println("")
   return nothing
   elseif op == 2
   v4 = 12
   v4 = parse(Int64, v4)
   println("")
   println("Enter how many square meters the roof has: ")
   v3 = readline()
   v3 = parse(Int64, v3)
   println("")
   result2 = v3 * v4
   println("")
   println("The quantity of American tiles will be: ", result2)
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`)
   return nothing
   elseif op == 3
   v6 = 16
   v6 = parse(Int64, v6)
   println("")
   println("Enter how many square meters the roof has: ")
   v5 = readline()
   v5 = parse(Int64, v5)
   println("")
   result3 = v5 * v6
   println("")
   println("The quantity of colonial tiles will be: ", result3)
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`)
   return nothing
   elseif op == 4
   v8 = 14
   v8 = parse(Int64, v8)
   println("")
   println("Enter how many square meters the roof has: ")
   v7 = readline()
   v7 = parse(Int64, v7)
   println("")
   result4 = v7 * v8
   println("")
   println("The quantity of Italian tiles will be: ", result4)
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`)
   return nothing
   elseif op == 5
   v10 = 17
   v10 = parse(Int64, v10)
   println("")
   println("Enter how many square meters the roof has: ")
   v9 = readline()
   v9 = parse(Int64, v9)
   println("")
   result5 = v9 * v10
   println("")
   println("The quantity of Portuguese tiles will be: ", result5)
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`)
   return nothing
   elseif op == 6
   v12 = 16
   v12 = parse(Int64, v12)
   println("")
   println("Enter how many square meters the roof has: ")
   v11 = readline()
   v11 = parse(Int64, v11)
   println("")
   result6 = v11 * v12
   println("")
   println("The quantity of Roman tiles will be: ", result6)
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`)
   return nothing
   elseif op == 7
   println("Finished Algorithm!")
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`) 
   return nothing
   elseif op == 8
   run(`clear`)
   println("")
   println("Info")
   println("")
   println("To calculate the square meter of the roof we use Length x Width. [4 equal sides]")
   println("To calculate the Quantity of Tiles per Square Meter: Taking as an example an American tile with dimensions (43Lx26W) in centimeters in horizontal axis view,")
   println(" and knowing that calculating a square meter of a roof will be L x W then 1 Square Meter = 12 tiles, so a square meter has 12 tiles so this will be the standard measurement. 12 x so many square meters = the amount of tiles per square meter.")
   println("")
   println("To calculate Colonial Tiles: 1 Square Meter = 16 tiles.")
   println("To calculate Italian Tiles: 1 Square Meter = 14 tiles.")
   println("To calculate Portuguese Tiles: 1 Square Meter = 17 tiles.")
   println("")
   println("To calculate the Roman Tile: Taking as an example a Roman tile with dimensions (40Lx21W) in centimeters in horizontal axis view, and knowing that calculating a square meter of a roof will be L x W then 1 Square Meter = 16 tiles,")
   println(" so one square meter has 16 tiles, so this will be the standard measurement. 16 x so many square meters = the number of tiles per square meter.")
   println("")
   println("Important information: ")
   println("")
   println("This algorithm was built with integer variables so it does not accept numbers with commas e.g.: 2.90 meters change to 3 meters.")
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`) 
   return nothing
   elseif op == 9
   run(`clear`)
   println(" ") 
   println("Algorithm: Measurement and Calculation for Residential Roof") 
   println(" ")
   println("Author: PHNO") 
   println("Release Date: 18/11/2024") 
   println("Version: 1.0.1v")
   println("Replit: @PHNO, @REPLIT") 
   println("E-mail: phreplit@gmail.com ") 
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`) 
   return nothing
   else
   println("Error. Wrong Choice!")
   println(" ")
   println("Press enter to exit...")
   enter = readline()
   println(" ")
   run(`clear`) 
   return nothing
end

File: Algorithm - Measurement and Calculation for Residential Roof.m

#  Author: PHNO - Tecnólogo | Pós-Graduado
#  Release Date: 18/11/2024
#  Version: 1.0.1v
#  Replit: @PHNO, @PHREPLIT
#  E-mail: phreplit@gmail.com

# Algorithm: Measurement and Calculation for Residential Roof, with menu and console compilation.

fprintf('\n');
fprintf(' Measurement and Calculation for Residential Roof \n');
fprintf('\n');
fprintf('\n');
fprintf('1 - Calculate Square Meter of Roof [4 equal sides]\n');
fprintf('2 - Calculate the quantity of American tiles per Square Meter.\n');
fprintf('3 - Calculate the quantity of colonial tiles per Square Meter.\n');
fprintf('4 - Calculate the quantity of Italian tiles per Square Meter.\n');
fprintf('5 - Calculate the quantity of Portuguese tiles per Square Meter.\n');
fprintf('6 - Calculate the quantity of Roman tiles per Square Meter.\n');
fprintf('7 - Exit\n');
fprintf('[8] Info\n');
fprintf('[9] About\n');
fprintf('\n');

op = input('Enter with your choice: '); # choice - option
if (op == 1)
    fprintf('\n');
    v1 = int32(input('Enter the roof length: '));
    fprintf('\n');
    v2 = int32(input('Enter the roof width: '));
    result = (v1 * v2);
    fprintf('\n');
    fprintf('Square meters are equivalent to: %d square meters.\n', result);
    fprintf('\n');
    exit;
 elseif (op == 2)
    fprintf('\n');
    v4 = int32(12);
    fprintf('\n');
    v3 = int32(input('Enter how many square meters the roof has: '));
    fprintf('\n');
    result2 = v3 * v4;
    fprintf('');
    fprintf('The quantity of American tiles will be %d tile(s).\n', result2);
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
 elseif (op == 3)
    fprintf('\n');
    v6 = int32(16);
    fprintf('\n');
    v5 = int32(input('Enter how many square meters the roof has: '));
    fprintf('\n');
    result3 = v5 * v6;
    fprintf('');
    fprintf('The quantity of colonial tiles will be %d tile(s).\n', result3);
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
 elseif (op == 4)
    fprintf('\n');
    v8 = int32(14);
    fprintf('\n');
    v7 = int32(input('Enter how many square meters the roof has: '));
    fprintf('\n');
    result4 = v7 * v8;
    fprintf('');
    fprintf('The quantity of Italian tiles will be %d tile(s).\n', result4);
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
 elseif (op == 5)
    fprintf('\n');
    v10 = int32(17);
    fprintf('\n');
    v9 = int32(input('Enter how many square meters the roof has: '));
    fprintf('\n');
    result5 = v9 * v10;
    fprintf('');
    fprintf('The quantity of Portuguese tiles will be %d tile(s).\n', result5);
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
 elseif (op == 6)
    fprintf('\n');
    v12 = int32(16);
    fprintf('\n');
    v11 = int32(input('Enter how many square meters the roof has: '));
    fprintf('\n');
    result6 = v11 * v12;
    fprintf('');
    fprintf('The quantity of Roman tiles will be %d tile(s).\n', result6);
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
elseif (op == 7)
    fprintf('\n');
    fprintf('\nFinished Algorithm!');
    fprintf('\n');
    close;
    exit;
elseif (op == 8)
    clc; 
    fprintf('\n');
    fprintf('Info\n');
    fprintf('\n');
    fprintf('To calculate the square meter of the roof we use Length x Width. [4 equal sides]\n');
    fprintf('To calculate the Quantity of Tiles per Square Meter: Taking as an example an American tile with dimensions (43Lx26W) in centimeters in horizontal axis view,\n');
    fprintf(' and knowing that calculating a square meter of a roof will be L x W then 1 Square Meter = 12 tiles, so a square meter has 12 tiles so this will be the standard measurement. 12 x so many square meters = the amount of tiles per square meter.\n');
    fprintf('\n');
    fprintf('To calculate Colonial Tiles: 1 Square Meter = 16 tiles.\n');
    fprintf('To calculate Italian Tiles: 1 Square Meter = 14 tiles.\n');
    fprintf('To calculate Portuguese Tiles: 1 Square Meter = 17 tiles.\n');
    fprintf('\n');
    fprintf('To calculate the Roman Tile: Taking as an example a Roman tile with dimensions (40Lx21W) in centimeters in horizontal axis view, and knowing that calculating a square meter of a roof will be L x W then 1 Square Meter = 16 tiles,\n');
    fprintf(' so one square meter has 16 tiles, so this will be the standard measurement. 16 x so many square meters = the number of tiles per square meter.\n');
    fprintf('\n');
    fprintf('Important information: \n');
    fprintf('\n');
    fprintf('This algorithm was built with integer variables so it does not accept numbers with commas e.g.: 2.90 meters change to 3 meters.\n');
    fprintf(' '); 
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
elseif (op == 9)
    clc; 
    fprintf('\n');
    fprintf('\nAlgorithm: Measurement and Calculation for Residential Roof ');
    fprintf('\n');
    fprintf('\nAuthor: PHNO');
    fprintf('\nRelease Date: 18/11/2024');
    fprintf('\nVersion: 1.0.1v');
    fprintf('\nReplit: @PHNO, @PHREPLIT');
    fprintf('\nE-mail: phreplit@gmail.com');
    fprintf('\n');
    fprintf('\nPress enter to exit...');
    M = int32(input(' ')); 
    fprintf('\n');
    clc; 
    close;
    exit;
 elseif (op >= 10)
   fprintf('\n');
   fprintf('\nError. Wrong Choice!');
   fprintf('\n');
   exit;
 else
   fprintf('\n');
   fprintf('Error. choose again!');
   fprintf('\n');
   exit;
 end;
